mfs/dir.go

```diff
@@ -339,15 +339,25 @@ func (d *Directory) AddChild(name string, nd *dag.Node) error {
 
 	d.modTime = time.Now()
 
-	if len(nd.Links) == 0 {
+	pbn, err := ft.FromBytes(nd.Data)
+	if err != nil {
+		log.Warning("child was not unixfs node")
+		return err
+	}
+
+	switch pbn.GetType() {
+	case ft.TDirectory:
+		ndir := NewDirectory(d.ctx, name, nd, d, d.dserv)
+		d.childDirs[name] = ndir
+	case ft.TFile, ft.TMetadata, ft.TRaw, ft.TSymlink:
```

Comment:
I believe this is consistent with what the old code did -- what should happen instead?